src/ol_infrastructure/substructure/aws/eks/karpenter.py

```diff
@@ -392,3 +437,79 @@ def setup_karpenter(  # noqa: PLR0913
             depends_on=[karpenter_release],
         ),
     )
+
+    gpu_node_labels = {**k8s_global_labels, "ol.mit.edu/gpu_node": "true"}
+
+    # Karpenter will select this NodePool for pods with a node selector matching
+    # "ol.mit.edu/gpu_node": "true". When a pod is unschedulable, Karpenter evaluates
+    # all NodePools to find one that can satisfy the pod's scheduling constraints
+    # (like node selectors, taints, and resource requests). Because this NodePool's
+    # template adds the required label to new nodes, Karpenter will choose it over
+    # the default NodePool for matching pods.
+    kubernetes.apiextensions.CustomResource(
+        f"{cluster_name}-karpenter-gpu-node-pool",
+        api_version="karpenter.sh/v1",
+        kind="NodePool",
+        metadata=kubernetes.meta.v1.ObjectMetaArgs(
+            name="gpu",
+            namespace="operations",
+            labels=k8s_global_labels,
+        ),
+        spec={
+            "template": {
+                "metadata": {
+                    "labels": gpu_node_labels,
+                },
+                "spec": {
+                    "nodeClassRef": {
+                        "group": "karpenter.k8s.aws",
+                        "kind": "EC2NodeClass",
+                        "name": "default",
+                    },
+                    "taints": [
+                        {
+                            "key": "ol.mit.edu/gpu_node",
+                            "value": "true",
+                            "effect": "NoSchedule",
+                        }
+                    ],
+                    "expireAfter": "24h",
+                    "terminationGracePeriod": "30m",
+                    "requirements": [
+                        {
+                            "key": "node.kubernetes.io/instance-type",
+                            "operator": "In",
+                            "values": ["g4dn.xlarge", "g4dn.2xlarge"],
```

Comment:
It would be good to move this into an Enum that we can manage in the `lib/` module space for evolving the instance types. Similarly for the other node types that we want set in the default nodepool.